graphql/src/datasources/es/api/getQueryResults.ts

```diff
@@ -0,0 +1,315 @@
+  Extract<ElasticSearchIndex, typeof ES_EVENT_INDEX | typeof ES_LOCATION_INDEX>,
+  SearchResultField
+> = {
+  [ES_EVENT_INDEX]: EVENT_SEARCH_RESULT_FIELD,
+  [ES_LOCATION_INDEX]: VENUE_SEARCH_RESULT_FIELD,
+};
+
+const buildMustHaveReservableResourceFilter =
+  (): MustHaveReservableResourceFilter => ({
+    bool: {
+      should: [
+        // i.e. OR
+        {
+          term: {
+            'venue.resources.reservable': true,
+          },
+        },
+        {
+          exists: {
+            field: 'venue.resources.externalReservationUrl',
+          },
+        },
+      ],
+    },
+  });
+
+export type getQueryResultsProps = {
+  text?: string;
+  ontology?: string;
+  administrativeDivisionIds?: string[];
+  ontologyTreeIdOrSets?: string[][];
+  ontologyWordIdOrSets?: string[][];
+  providerTypes?: string[];
+  serviceOwnerTypes?: string[];
+  targetGroups?: string[];
+  mustHaveReservableResource?: boolean;
+  index?: ElasticSearchIndex;
+  from?: number;
+  size?: number;
+  languages?: ElasticLanguage[];
+  openAt?: string;
+  orderByDistance?: OrderByDistanceParams;
+  orderByName?: OrderByNameParams;
+  orderByAccessibilityProfile?: AccessibilityProfileType;
+};
+
+export default async function getQueryResults(
+  request: ElasticSearchAPI['post'],
+  {
+    text,
+    ontology,
+    administrativeDivisionIds,
+    ontologyTreeIdOrSets,
+    ontologyWordIdOrSets,
+    providerTypes,
+    serviceOwnerTypes,
+    targetGroups,
+    mustHaveReservableResource,
+    index,
+    from,
+    size,
+    languages,
+    openAt,
+    orderByDistance,
+    orderByName,
+    orderByAccessibilityProfile,
+  }: getQueryResultsProps
+) {
+  const es_index = index || ES_DEFAULT_INDEX;
+
+  // Some fields should be boosted / weighted to get more relevant result set
+  const searchFieldsBoostMapping = {
+    // Normally weighted search fields for different indexes
+    1: (lang: ElasticLanguage, index: ElasticSearchIndex) => {
+      return (
+        {
+          [ES_LOCATION_INDEX]: [`venue.description.${lang}`],
+          [ES_EVENT_INDEX]: [`event.name.${lang}`, `event.description.${lang}`],
+        }[index] ?? []
+      );
+    },
+    3: (lang: ElasticLanguage, index: ElasticSearchIndex) => {
+      return (
+        {
+          [ES_LOCATION_INDEX]: [`venue.name.${lang}`],
+        }[index] ?? []
+      );
+    },
+  };
+
+  // Ontology fields for different indexes
+  const ontologyFields = (lang: ElasticLanguage, index: ElasticSearchIndex) => {
+    if (index === ES_LOCATION_INDEX) {
+      return [
+        `links.raw_data.ontologyword_ids_enriched.extra_searchwords_${lang}`,
+        `links.raw_data.ontologyword_ids_enriched.ontologyword_${lang}`,
+        `links.raw_data.ontologytree_ids_enriched.name_${lang}`,
+        `links.raw_data.ontologytree_ids_enriched.extra_searchwords_${lang}`,
+      ];
+    } else if (index === ES_EVENT_INDEX) {
+      return [`ontology.${lang}`, 'ontology.alt'];
+    }
+    return [];
+  };
```

Comment:
Should be moved outside the function, so the dependencies and the context becomes clearer.